File: martix_impl/src/main/java/com/xm/martix_impl/matrix/TestPluginListener.java
package com.xm.martix_impl.matrix;

import android.annotation.SuppressLint;
import android.content.Context;
import android.util.Log;

import com.tencent.matrix.plugin.DefaultPluginListener;
import com.tencent.matrix.report.Issue;

public class TestPluginListener extends DefaultPluginListener {
    public static final String TAG = "Matrix.TestPluginListener";

    public TestPluginListener(Context context) {
        super(context);

    }

    @SuppressLint("LongLogTag")
    @Override
    public void onReportIssue(Issue issue) {
        super.onReportIssue(issue);
        Log.e(TAG, issue.toString());
        //add your code to process data
    }
}

File: martix_impl/src/main/java/com/xm/martix_impl/matrix/TestPluginListener.kt
package com.xm.martix_impl.matrix

import android.annotation.SuppressLint
import android.app.Application
import android.content.Context
import android.util.Log
import com.tencent.matrix.plugin.DefaultPluginListener
import com.tencent.matrix.report.Issue
import java.io.File
import java.text.SimpleDateFormat
import java.util.*

class TestPluginListener(var context: Context?) : DefaultPluginListener(context) {
    companion object {
        const val TAG = "Matrix.TestPluginListener"
    }

    @SuppressLint("LongLogTag")
    override fun onReportIssue(issue: Issue) {
        super.onReportIssue(issue)
        Log.e(TAG, issue.toString())
        logInFile(issue.toString())
    }

    private fun logInFile(txt: String) {
        try {
            val application = context as Application
            val format = SimpleDateFormat("yyyy-MM-dd-HH:mm:ss")
            val timeStr = format.format(Date())
            val cacheDir = File("${application.cacheDir}/matrix_block_log")
            val file = File(cacheDir, "${timeStr}.txt")
            if (cacheDir.exists().not()) {
                cacheDir.mkdirs()
            }
            file.writeText(txt)
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }
}